### ISSUE TITLE
Fully Hide Follower(s) and content inset issues

### ISSUE THREAD

**Issue by StainlessStlRat** [andreamazz/AMScrollingNavbar#335]
**Describe the bug**
I am replacing TLYShyNavBarManager with your library and liking it very much! However, I'm not sure I understand how to to use it correctly.

I have a UIViewController that has a full screen tableView.
I create a custom toolbar programmatically and add it as a subview to self.view (I suspect herein lies my content inset issue.  With TLY, I didn't have to add it to self.view)

Everything works GREAT except:

1) The toolbar I have set as a follower does not fully hide.  I can actually live with this until a future update.
2) The toolbar covers up some of the underlying tableview.  This I have to fix ASAP.

Relevant code:

//Creating of toolbar
_statusBar = [[StatusBar alloc] initWithFrame:CGRectMake(0.0, 0.0, self.view.frame.size.width, 130.0)];
_statusBar.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
_statusBar.searchBar.delegate = self; //Implement search bar delegate features
[self.view addSubview:_statusBar];

//Attaching/Following
NavigationBarFollower *statusBarFollow = [[NavigationBarFollower alloc] initWithView:_statusBar direction:NavigationBarFollowerCollapseDirectionScrollUp];
[(ScrollingNavigationController *)self.navigationController followScrollView:_tableView delay:0 scrollSpeedFactor:1 collapseDirection:NavigationBarCollapseDirectionScrollDown followers:@[statusBarFollow]];

Here is a video demonstrating my issue!
https://imgur.com/a/hWmjgrf

Any help would be greatly appreciated! And thanks for the library!

**Comment by StefaniOSApps** [comment]
Do **not** use 

```
if (@available(iOS 11.0, *)) {
          [table setContentInsetAdjustmentBehavior:UIScrollViewContentInsetAdjustmentNever];
}
```

**Comment by StainlessStlRat** [comment]
Thanks Stefani, I already checked and I am not using that (though that was being called in the old TLYShyNav lib)

**Comment by StefaniOSApps** [comment]
the best way is to make a sample project and upload it on GitHub. Yes, this framework has some issues, including the gaps. I have now noticed a new issue, then I will post with a demo project.

**Comment by StainlessStlRat** [comment]
I will gladly create a demo project if it is helpful. Just want to make sure I’m not doing something wrong.

**Comment by StefaniOSApps** [comment]
That will be the best. You can also test my creation of the framework. There are some errors already fixed. Also make sure you have set translucens to true.

**Comment by StainlessStlRat** [comment]
I ran your demo last night. For whatever reason for me the performance was very bad, even in release mode. Ill try again tonight.

**Comment by StefaniOSApps** [comment]
In the Demo Projet I have outsourced the framework from cocoapod. Only so could see in messages of the terminal.

https://github.com/StefaniOSApps/Example

or use

```
pod 'AMScrollingNavbar',
  :git => 'https://github.com/StefaniOSApps/AMScrollingNavbar.git'#,
  #:branch => 'master'
```

**Comment by StainlessStlRat** [comment]
https://github.com/StainlessStlRat/ScrollBarDemo

It doesn't behave precisely like my app does during integration, but close enough to demonstrate the issue with the gap and not hiding the follower.

Please let me know if there is anything else I can do to help.

**Comment by StefaniOSApps** [comment]
the content inset can you see here:
<img width="623" alt="bildschirmfoto 2018-10-19 um 21 57 44" src="https://user-images.githubusercontent.com/29314643/47241012-75fb1f80-d3ea-11e8-88f2-2794646d63c3.png">

I think the hight of UITableView is fix.

**Comment by StainlessStlRat** [comment]
How can I go about fixing that? Shouldn’t it be an offset/inset that is passed in by the scroll manager?

Also what about the follower not hiding?

**Comment by StainlessStlRat** [comment]
And how do you look at the interface that way? :)

**Comment by StainlessStlRat** [comment]
Here is an example of the functionality I need to replace (old version of my app integrated with TLYShyNavBar)
https://imgur.com/a/UTs1HGf

You can see the follower hides first, then the nav bar.

**Comment by StefaniOSApps** [comment]
solved on https://github.com/StefaniOSApps/Example/issues/1#issuecomment-431606120

**Comment by StainlessStlRat** [comment]
Hello, just checking in again. 

If the author gets a chance, can you let me know whether or not this is a bug or if it’s just something that the library wasn’t designed for etc...

I can understand if you don’t have time to fix it, but just understanding whether or not it is truly a bug would be helpful.

**Comment by andreamazz** [comment]
Let me check

**Comment by andreamazz** [comment]
Ok, I'm reading back the issue, in the end your request is, if I understand corectly, how to implement this behaviour: https://imgur.com/a/UTs1HGf

The `followers` feature works th other way around, they follow the navbar while it's scrolling away, but they remain on screen at all times. 
On top of my head I think you can recreate that effect by serving the header view inside the cell at indexpath 0,0, and use the scroll delay feature to account for its height.

**Comment by andreamazz** [comment]
Here, like this:
[ScrollBarDemo.zip](https://github.com/andreamazz/AMScrollingNavbar/files/2623754/ScrollBarDemo.zip)

**Comment by StainlessStlRat** [comment]
Hey thank you andremazz!! I will look at your demo and look at replacing the one I use with yours again.  I realllllllly hope I can get it working because the library I'm using is no longer in development.

**Comment by StainlessStlRat** [comment]
Hello andreamazz, I checked out your demo but unfortunately it's not quite what I'm after.  If you look closely at the Imgur gif, the toolbar underneath the navbar can be visible at any index.  So you could be at cell 500 for example, pull down a little and the toolbar will still pop up.

If you have a moment, maybe you can download my app @ https://itunes.apple.com/us/app/gameye/id1105342771?mt=8
and see how it currently works to see if it's possible to implement with your library?

Just open the app, click on games, and go to the encyclopedia tab and scroll around.  I would absolutely LOVE to replace it's implementation with yours, and if this functionality isn't currently in the library, I think your users would find it quite useful!

**Comment by andreamazz** [comment]
Ok, I see what you mean, let me think about it.
Btw, the scrolling sometimes is janky on an iPhone XS, it feels like the app is calling networking code on the main thread. Out of curiosity, what are you using to download images?

**Comment by StainlessStlRat** [comment]
So I finally figured out that the library I have been using is occasionally crashing via OOM, which Crashylytics doesn't catch. Hopefully you figure something out!

**Comment by andreamazz** [comment]
Yep, I'm working on this feature. It's not straightforward to implement, it might take a while.

**Comment by StainlessStlRat** [comment]
That’s great news!!!! I was able to patch the oom, but it’s hack on hack on hack so I cant wait for your version!

**Comment by StainlessStlRat** [comment]
I’m glad youre working on it!!

**Comment by andreamazz** [comment]
I'm a bit swamped between work and moving from my old apartment. I can push the current state on a separate branch, any help is welcome.

**Comment by andreamazz** [comment]
Checkout branch `header`

**Comment by X901** [comment]
Any update ?
I faced same issue 

https://cl.ly/f272f705d050

**Comment by StainlessStlRat** [comment]
Any progress? I don't mind looking at the branch, but is there certain issues you're facing?

**Comment by StainlessStlRat** [comment]
Any updates?

**Comment by StainlessStlRat** [comment]
Just checking in to see if there’s any progress?

**Comment by andreamazz** [comment]
Hey @StainlessStlRat 
sorry for the delay, I'm currently unable to work on this. There is a partial implementation in the `header` branch, any help is welcome.
